### ISSUE TITLE
Datagrid: Reorderable columns

### ISSUE THREAD

**Issue by r-bowen** [vmware/clarity#145]
```
[ ] bug
[x ] feature request
[ ] enhancement
```

### Expected behavior
The column width of the datagrid is controlled by the size of the data in the cells. It would be really useful for the user to have control over the column width by dragging the column headers.

It would also be really useful if the user could also drag and reorder the columns.

These changes will make data analysis much easier for the user.

### Environment details

* **Angular version:** 2.1.0

* **Clarity version:** 0.7.0

* **OS and version:** 

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

**Comment by youdz** [comment]
We have a design for "column selection", meaning the user gets to choose which columns are displayed. 

This, however, is a different feature that needs to go through the UX team for evaluation and a first design. We can't give you a timeline on this yet, we need to evaluate the priority compared to other Datagrid enhancements.

Thank you for the suggestion.

**Comment by jaffoneh** [comment]
Closing since it will be addressed in: #621 so this becomes a duplicate. Thanks for filing and hope to see it land in Clarity soon!

**Comment by jaffoneh** [comment]
Actually, @Shijir just pointed out that re-ordering columns won't land with #621 so reopening this ticket to keep track of that request.

**Comment by mathisscott** [comment]
We need to understand use cases to determine priority and difference between this request and displaying columns in the most logical order.

**Comment by reddolan** [comment]
We discussed this and won't be implementing sortable columns. Here are the reasons why:
- If we enabled this, we make it easy for teams to put the responsibility on users to sort things. Teams should be doing their diligence to know the best order for the columns of a datagrid
- even if teams aren't able research to know what the best order is, they should be able to change it easily based on user feedback

**Comment by bdunkz** [comment]
Hi @reddolan - we have cases where screens are used by many people with different roles.  Due to this, a set of columns that may be most interesting to one user may not be for another.  By allowing the user to reorder them, we would be able to address the needs of users working in different roles.  Can we take another pass at evaluating this?  Thanks.

**Comment by mathisscott** [comment]
@reddolan 
Based on @bdunkz use case, do you see this as a viable feature to include in Clarity?

**Comment by reddolan** [comment]
@mathisscott & @bdunkz, yes this case makes sense to have the ability to rearrange columns. We will also need to be explicit in docs on how it should be used to keep from being used everywhere and unnecessarily weighing down page experiences.

**Comment by youdz** [comment]
I think we're missing one of the main reasons this was closed.
The UX team decided at the time that having both resizable columns and hide/show capabilities on columns allowed the user to display exactly the relevant information on the screen at all times. As such, it served exactly the same use case as reorderable columns, and introducing all 3 features just bloated the datagrid design for duplicate solutions.

In other words, we had:
- Resizable columns
- Hide/show columns
- Reorderable columns
We picked 2 out of 3, because all 3 would be redundant.

If we decide to go back on this so be it, but I would like previous research and discussion to be considered in the decision, rather than pile on more features because why not.

**Comment by youdz** [comment]
@reddolan: We're trying to figure out if this is actually work needed for 1.0 or not. Any news on what the plan is?

Even if the answer is "we haven't looked over the research yet", at least we'll know to triage this as a UX task.

**Comment by marvinosswald** [comment]
@mathisscott any update here ?

**Comment by mathisscott** [comment]
Hi @marvinosswald 
The first piece of this was creating a general drag-and-drop directive. That's [in review now](https://github.com/vmware/clarity/issues/1994).

After that is completed, we will move on to applying drag-and-drop to the columns of the datagrid.

**Comment by namitabandal** [comment]
Any updates on column reordering feature ?
